src/display/api.js

```diff
@@ -2053,11 +2054,13 @@ var WorkerTransport = (function WorkerTransportClosure() {
     },
 
     getMetadata: function WorkerTransport_getMetadata() {
+      let fullReader = this._fullReader;
       return this.messageHandler.sendWithPromise('GetMetadata', null).
         then(function transportMetadata(results) {
         return {
           info: results[0],
           metadata: (results[1] ? new Metadata(results[1]) : null),
+          attachedFileName: fullReader.fileName,
```

Comment:
I'm a bit confused about the name `attachedFileName` because it has nothing to do with attachments in PDF files, yet that's what I think about first when I read it. Would `headerFileName` or `contentDispositionFileName` perhaps be better? If so, do not forget to change this elsewhere in the patch too.